File: English_back_java/src/main/java/com/austain/srevice/ProfileService.java
package com.austain.srevice;

import com.austain.domain.dto.profile.ProfileDashboardDTO;
import com.austain.domain.dto.profile.ProfileDashboardDTO.Settings;
import com.austain.domain.dto.profile.ProfileDashboardDTO.PasswordUpdate;

public interface ProfileService {
    ProfileDashboardDTO getProfile(Long userId);

    Settings updateSettings(Long userId, Settings settings);

    void updatePassword(Long userId, PasswordUpdate passwordUpdate);
}

File: English_front_vue/src/services/profile.service.js
import api from './api'

export const profileService = {
  getProfile() {
    return api.get('/api/profile/me')
  },

  updateSettings(payload) {
    return api.put('/api/profile/settings', payload)
  },

  updatePassword(payload) {
    return api.put('/api/profile/password', payload)
  },

  uploadAvatar(file) {
    const formData = new FormData()
    formData.append('file', file)
    return api.post('/api/profile/avatar/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    })
  }
}
